src/components/Header/index.jsx

```diff
@@ -1,102 +1,120 @@
+        <Dropdown.Item
+          href={getLogoutRedirectUrl()}
+        >
+          Logout
+        </Dropdown.Item>
+      </Dropdown.Menu>
+    </Dropdown>
+  );
+};
+
+HeaderDropdown.propTypes = {
+  user: PropTypes.shape({
+    profileImage: PropTypes.shape({
+      hasImage: PropTypes.bool.isRequired,
+      imageUrlMedium: PropTypes.string,
+    }),
+    username: PropTypes.string.isRequired,
+  }).isRequired,
+};
+
+const Header = ({
+  hasSidebarToggle, enterpriseName, enterpriseLogo,
+}) => {
+  const user = getAuthenticatedUser();
+  useEffect(() => {
+    // if there is no email, the user data has not been hydrated
+    if (user && !user.email) {
+      hydrateAuthenticatedUser();
     }
-    return <Icon className="fa fa-user mr-2" screenReaderText={screenReaderText} />;
-  }
-
-  renderLogo() {
-    const { enterpriseLogo, enterpriseName } = this.props;
-    const logo = configuration.LOGO_URL;
-    return (
-      <Img
-        src={enterpriseLogo || logo}
-        alt={`${enterpriseName || 'edX'} logo`}
-        onError={(e) => { e.target.src = logo; }}
-      />
-    );
-  }
-
-  render() {
-    const { email, hasSidebarToggle } = this.props;
-    return (
-      <header className="container-fluid border-bottom">
-        <nav aria-label="header" className="navbar px-0 py-1 justify-content-between">
-          <div>
-            {hasSidebarToggle && <SidebarToggle />}
-            <Link
-              to="/"
-              className="navbar-brand"
-            >
-              {this.renderLogo()}
-            </Link>
-          </div>
-          {email && (
-            // TODO: @edx/paragon's Dropdown is now a passthru component from react-bootstrap
-            // that doesn't address focus management, so using the now-deprecated
-            // previous version of the Dropdown component until the a11y issue is fixed.
-            <Dropdown.Deprecated>
-              <Dropdown.Deprecated.Button className="btn-outline-primary">
-                {this.getProfileIconElement()}
-                {email}
-              </Dropdown.Deprecated.Button>
-              <Dropdown.Deprecated.Menu>
-                <Button
-                  variant="link"
-                  className="dropdown-item"
-                  onClick={() => apiClient.logout()}
-                >
-                  Logout
-                </Button>
-              </Dropdown.Deprecated.Menu>
-            </Dropdown.Deprecated>
-          )}
-        </nav>
-      </header>
-    );
-  }
-}
+    // rehydrate if the username changes
+  }, [user?.username, user?.id]);
+
+  return (
+    <header className="container-fluid border-bottom">
+      <Navbar aria-label="header" className="px-0 py-1 justify-content-between">
+        <Nav>
+          {hasSidebarToggle && <SidebarToggle />}
+
+          <Nav.Link
+            href="/"
+            className="navbar-brand"
+          >
+            <Logo enterpriseLogo={enterpriseLogo} enterpriseName={enterpriseName || ''} />
```

Comment:
nit: Is the ` || ''` needed for the `enterpriseName` prop, or could `Logo.defaultProps` default it to the empty string?

Reply:
Aaaah it was `null` in the defaultProps and I think that was causing issues. Updating.